File: Graph/ScriptTemplate/GreaterEqual/CGGreaterEqual.lua
local CGGreaterEqual = CGGreaterEqual or {}
CGGreaterEqual.__index = CGGreaterEqual

function CGGreaterEqual.new()
    local self = setmetatable({}, CGGreaterEqual)
    self.inputs = {}
    return self
end

function CGGreaterEqual:setInput(index, func)
    self.inputs[index] = func
end

function CGGreaterEqual:getOutput(index)
    return self.inputs[0]() >= self.inputs[1]()
end

return CGGreaterEqual

File: Graph/ScriptTemplate/GreaterEqual/CGGreaterEqual.js
/**
 * @file CGGreaterEqual.js
 * @author
 * @date 2021/8/15
 * @brief CGGreaterEqual.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGGreaterEqual extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return this.inputs[0]() >= this.inputs[1]();
  }
}

exports.CGGreaterEqual = CGGreaterEqual;
